### DISCUSSION TITLE
Finalisation of HA set up

### DISCUSSION THREAD

**Discussion by Loudo42** [cmcgerty/Sofar2mqtt#35] [Q&A]
Hi,

I am stopped at the end of the set up. I am new in HA and need some help.

Here is my config:
Inverter HYD-6000-EP with 4 batteries US2000
Raspeberry Pi4 with raspbian, home assistant supervised with docker
Home assistant add-on for Mosquitto and zigbee2mqtt and Node-red.

I have flashed the ESP8266 and it is well connected to my Wifi and Mqqt broker. I have well the alternate RS485 error and CRC-fault message. But I do not see entities in HA.

I guess it is because I do not know how to subscribe my Mqtt client to sofar2mqtt/state.
Where have I to implement this line of code, in the integration page or in the add-on or in the configuration.yaml?

How send a command to test by mqqt-pub to check if it works?
What is the exact syntaxe, can somebody share a screen-shot?

To have open energy in node-red, have I to follow this installation
[https://www.home-assistant.io/integrations/emoncms/]
, I feel it strange because it is an implementation in HA and not Node-red, will it work?
If not could you show me the way.

Like described in the discussion  ''Integration in Home Assistant #26'' I have and HYD-6000-EP inverter, so I am certainly not near to succeed :(

Before trying sofar2mqtt, I have tried solarman integration, but no luck my logger is not supported, and here I was really very interested to get real-time data.

Thank you very much in advance for your help.

Ludo

**Comment by MartinJSa** [comment]
I've created a pull request for my HA configuration.yaml to add to this project, hopefully that'll be accepted soon and you can see how I configured HA to work with MQTT (no need for Node Red).

M.

**Comment by cmcgerty** [comment]
Thanks Martin.

Merged and added to the overdue v2.0 release.

Colin


On 5 Nov 2022, at 17:17, Martin Saunders ***@***.***> wrote:

﻿

I've created a pull request for my HA configuration.yaml to add to this project, hopefully that'll be accepted soon and you can see how I configured HA to work with MQTT (no need for Node Red).

M.

**Reply by Loudo42** [reply]
Hi,

Thank you for the new version,

I succeed to collect the data by changing the register for my HYD-6000-EP (see discussion #26 Integration in Home assistant), but something is still wrong.

The LCD screen show all the time Normal, both in passive mode or Self use mode.
When I set up passive mode, nothing happen, the inverter do not charge neither discharge the battery. It means it does not set up the inverter in auto mode whereas it should.
When I set up self-use-mode, if the battery is charging, the value on the screen is correct, but while discharging, the value is not correct.

When I send command to set up auto or anything else, nothing happen, I have a RS485 error and CRC fault few second after sending.
If I listen Sofar2mqtt/response (or response/), I do not have any reply after sending a mqtt message.
If I listen Sofar2mqtt/#, I the following reply:
Response too short
QoS:0 - Retain False.

Somebody has a HYD - EP inverter and reach to send message?
Maybe the register for runstate is also different, or one of this:
#define MODBUS_FN_READSINGLEREG 0x03
#define SOFAR_FN_PASSIVEMODE    0x42
#define SOFAR_PARAM_STANDBY     0x5555
#define SOFAR_FN_STANDBY	0x0100
#define SOFAR_FN_DISCHARGE	0x0101
#define SOFAR_FN_CHARGE	0x0102
#define SOFAR_FN_AUTO	0x0103
Any ideas?

Thank you very much in advance,

Ludo
